File: frontend/src/app/features/admin/scheduler/data/scheduler.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../../../environments/environment';

export type DummyDataTable = 'Category' | 'Product' | 'Order';

export interface GenerateDummyDataRequest {
  table: DummyDataTable;
  targetRowCount?: number;
  targetStorageBytes?: number;
}

/** Matches Scheduler.Application.DTOs.JobRunDto. */
export interface JobRun {
  id: string;
  hangfireJobId: string;
  jobType: string;
  status: 'Processing' | 'Success' | 'Error';
  resultMessage: string | null;
  createdAt: string;
  completedAt: string | null;
}

@Injectable({ providedIn: 'root' })
export class SchedulerService {
  private readonly http = inject(HttpClient);

  generateDummyData(request: GenerateDummyDataRequest): Observable<{ hangfireJobId: string }> {
    return this.http.post<{ hangfireJobId: string }>(`${environment.apiUrl}/scheduler/generate-dummy-data`, request);
  }

  listJobs(): Observable<JobRun[]> {
    return this.http.get<JobRun[]>(`${environment.apiUrl}/scheduler/jobs`);
  }
}

File: backend/Modules/Scheduler/Scheduler.Infrastructure/Persistence/SchedulerService.cs
using Hangfire;
using Microsoft.EntityFrameworkCore;
using Scheduler.Application.Abstractions;
using Scheduler.Application.DTOs;
using Scheduler.Infrastructure.Entities;
using Scheduler.Infrastructure.Jobs;

namespace Scheduler.Infrastructure.Persistence;

public sealed class SchedulerService(SchedulerDbContext db, IBackgroundJobClient backgroundJobClient) : ISchedulerService
{
    public async Task<string> EnqueueGenerateDummyDataAsync(GenerateDummyDataRequest request, CancellationToken ct = default)
    {
        var jobRun = new JobRun
        {
            HangfireJobId = string.Empty,
            JobType = $"GenerateDummyData:{request.Table}",
            Status = "Processing"
        };
        db.JobRuns.Add(jobRun);
        await db.SaveChangesAsync(ct);

        var hangfireId = backgroundJobClient.Enqueue<GenerateDummyDataJob>(
            job => job.RunAsync(jobRun.Id, request.Table, request.TargetRowCount, request.TargetStorageBytes));

        jobRun.HangfireJobId = hangfireId;
        await db.SaveChangesAsync(ct);

        return hangfireId;
    }

    public async Task<IReadOnlyList<JobRunDto>> ListRecentJobsAsync(int take = 20, CancellationToken ct = default) =>
        await db.JobRuns.AsNoTracking()
            .OrderByDescending(j => j.CreatedAt)
            .Take(take)
            .Select(j => new JobRunDto(j.Id, j.HangfireJobId, j.JobType, j.Status, j.ResultMessage, j.CreatedAt, j.CompletedAt))
            .ToListAsync(ct);
}
